src/components/forms/SearchableLanguageSelector.tsx

```diff
@@ -0,0 +1,163 @@
+      <div
+        onClick={handleToggle}
+        className={`
+          w-full px-4 py-3 border border-gray-300 rounded-lg shadow-sm 
+          focus-within:outline-none focus-within:ring-2 focus-within:ring-blue-500 focus-within:border-transparent 
+          disabled:bg-gray-100 disabled:text-gray-600 disabled:cursor-not-allowed 
+          text-base cursor-pointer
+          ${disabled ? 'cursor-not-allowed' : ''}
+        `}
+      >
+        <div className="flex items-center justify-between">
+          <span className={value ? 'text-gray-900' : 'text-gray-500'}>
+            {selectedLanguageName}
+          </span>
+          <div className="flex items-center gap-2">
+            {value && !disabled && (
+              <button
+                type="button"
+                onClick={handleClear}
+                className="text-gray-400 hover:text-gray-600 focus:outline-none"
+                aria-label="Clear selection"
+              >
+                <X className="h-4 w-4" />
+              </button>
+            )}
+            <ChevronDown
+              className={`h-5 w-5 text-gray-400 transition-transform ${
+                isOpen ? 'transform rotate-180' : ''
+              }`}
+            />
+          </div>
+        </div>
+      </div>
+
+      {isOpen && !disabled && (
+        <div className="absolute z-10 w-full mt-1 bg-white border border-gray-200 rounded-lg shadow-lg">
+          <div className="p-2 border-b border-gray-200">
+            <input
+              ref={inputRef}
+              type="text"
+              value={searchQuery}
+              onChange={(e) => setSearchQuery(e.target.value)}
+              placeholder="Search languages..."
+              className="w-full px-3 py-2 border border-gray-300 rounded-md text-sm focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"
+              onClick={(e) => e.stopPropagation()}
+            />
+          </div>
+          <div className="max-h-60 overflow-y-auto">
+            <button
+              type="button"
+              onClick={() => handleSelect('')}
+              className={`
+                w-full px-4 py-2 text-left hover:bg-gray-50 focus:bg-gray-50 focus:outline-none
+                ${!value ? 'bg-blue-50 text-blue-900' : 'text-gray-900'}
+              `}
+            >
+              None
+            </button>
+            {filteredLanguages.length > 0 ? (
+              filteredLanguages.map((language) => (
+                <button
+                  key={language.code}
+                  type="button"
+                  onClick={() => handleSelect(language.code)}
+                  className={`
+                    w-full px-4 py-2 text-left hover:bg-gray-50 focus:bg-gray-50 focus:outline-none
+                    ${value === language.code ? 'bg-blue-50 text-blue-900' : 'text-gray-900'}
+                  `}
+                >
+                  {language.name}
```

Comment:
Just one tweak here I been thinking about, can we have the language display as `Plains Cree Y-dialect (crk)` as well? This will correspond to some changes I have planned for the documentation where we can list off some of the unique "idiosyncrasies" (is that the right word?) or details about various FSTs because they may not all necessarily behave exactly the same?